### ISSUE TITLE
Resultados del Autoencoder Convolucional - FamaF 2021

### ISSUE THREAD

**Issue by LUGOMauricioAlejandro** [LUGOMauricioAlejandro/Machine-Learning--FAMAF-2021#1]
# Visualización del Resultado del autoencoder 
import numpy as np
import matplotlib.pyplot as plt
plt.ion()

import torchvision.utils

autoencoder.eval()

# Esta función toma como entrada las imágenes a reconstruir y el nombre del modelo con el que se realizan las reconstrucciones se realizan las reconstrucciones

def to_img(x):
    x = 0.5 * (x + 1)
    x = x.clamp(0, 1)
    return x

def show_image(img):
    img = to_img(img)
    npimg = img.numpy()
    plt.imshow(np.transpose(npimg, (1, 2, 0)))

def visualise_output(images, model):

    with torch.no_grad():

        images = images.to(device)
        images = model(images)
        images = images.cpu()
        images = to_img(images)
        np_imagegrid = torchvision.utils.make_grid(images[1:4], 300, 1).numpy()
        plt.imshow(np.transpose(np_imagegrid, (1, 2, 0)))
        plt.show()

images, labels = iter(test_dataloader).next()

# Primero visualiza las imágenes originales
print('Imagen Original')
show_image(torchvision.utils.make_grid(images[1:4],300,1))
plt.show()

# Reconstruir y visualizar las imágenes con el autocodificador
print('Imagen Recosntruida :')
visualise_output(images, autoencoder)

# Imagen Original del Repo Fashion Mnist
![image](https://user-images.githubusercontent.com/20970851/147490573-d09b7067-0746-40b3-b6a0-53f061eccb68.png)
 # Imagen Recosntruida con el Autoencoder Convolucional 
![image](https://user-images.githubusercontent.com/20970851/147490722-911eea5b-f820-41a3-b0dd-823b92f3ff32.png)
